compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -1047,6 +1050,49 @@ private STToken processIdentifierOrKeyword() {
         }
     }
 
+    private int getNextNonWSOrNonCommentChar() {
+        int lookaheadCount = 0;
+        while (!reader.isEOF()) {
+            char c = reader.peek(lookaheadCount);
+            switch (c) {
+                case LexerTerminals.SPACE:
+                case LexerTerminals.TAB:
+                case LexerTerminals.FORM_FEED:
+                case LexerTerminals.CARRIAGE_RETURN:
+                case LexerTerminals.NEWLINE:
+                    lookaheadCount++;
+                    break;
+                case LexerTerminals.SLASH:
+                    if (reader.peek(lookaheadCount + 1) == LexerTerminals.SLASH) {
+                        lookaheadCount += 2;
+                        lookaheadCount = skipComment(lookaheadCount);
+                        break;
+                    }
+                    return reader.peek(lookaheadCount);
+                default:
+                    return reader.peek(lookaheadCount);
```

Comment:
Did we add the test cases to check this?

```bal
function main() {
    string:RegExp b = re // some ------------- comment -----------------  string
    ``;
}
```

Reply:
Let's add a few other cases as well

Reply:
I will add tests

Reply:
Updated 974f263b10627f4ee1addbe06e1808c4682ce2c7